source/dofs/dof_tools_constraints.cc

```diff
@@ -1070,6 +1101,1290 @@ namespace DoFTools
+                    continue;
+
+                  Assert(cell->face(f)->child(c)->n_active_fe_indices() == 1,
+                         ExcInternalError());
+                  // right now, all that is implemented is the case that both
+                  // sides use the same FE
+                  Assert(cell->face(f)->child(c)->fe_index_is_active(
+                           cell->active_fe_index()) == true,
+                         ExcNotImplemented());
+                }
+#endif // DEBUG
+
+              // Ok, start up the work:
+              // get the fe
+              const FiniteElement<dim, spacedim> &fe = cell->get_fe();
+
+              const unsigned int n_dofs = fe.n_dofs_per_line();
+              face_dof_indices.resize(n_dofs);
+
+              cell->face(f)->get_dof_indices(face_dof_indices);
+              const std::vector<types::global_dof_index> dof_on_mother_face =
+                face_dof_indices;
+
+              cell->face(f)->child(0)->get_dof_indices(face_dof_indices);
+              const std::vector<types::global_dof_index> dof_on_child_face_0 =
+                face_dof_indices;
+
+              cell->face(f)->child(1)->get_dof_indices(face_dof_indices);
+              const std::vector<types::global_dof_index> dof_on_child_face_1 =
+                face_dof_indices;
+
+              // get the direction of the faces
+              const int direction_mother = (cell->face(f)->vertex_index(0) >
+                                            cell->face(f)->vertex_index(1)) ?
+                                             1 :
+                                             -1;
+              const int direction_child_0 =
+                (cell->face(f)->child(0)->vertex_index(0) >
+                 cell->face(f)->child(0)->vertex_index(1)) ?
+                  1 :
+                  -1;
+              const int direction_child_1 =
+                (cell->face(f)->child(1)->vertex_index(0) >
+                 cell->face(f)->child(1)->vertex_index(1)) ?
+                  1 :
+                  -1;
+
+              // now enter a constraint that child face dofs are
+              // linear combinations of the parent face dofs
+              for (unsigned int row = 0; row < n_dofs; ++row)
+                {
+                  constraints.add_line(dof_on_child_face_0[row]);
+                  constraints.add_line(dof_on_child_face_1[row]);
+                }
+
+              for (unsigned int row = 0; row < n_dofs; ++row)
+                {
+                  for (unsigned int dof_i_on_mother = 0;
+                       dof_i_on_mother < n_dofs;
+                       ++dof_i_on_mother)
+                    {
+                      // We need to keep in mind that, if we use a FE_System
+                      // with multiple FE_NedelcSZ blocks inside, we need
+                      // to consider, that n_dofs depends on the number
+                      // of FE_NedelcSZ blocks used.
```

Comment:
```suggestion
                      // of FE_NedelecSZ blocks used.
```
I think there are a few more appearances of this kind, please fix them all.